src/lib/menu/menu-directive.ts

```diff
@@ -114,6 +123,28 @@ export class MatMenu implements AfterContentInit, MatMenuPanel, OnDestroy {
     this.setPositionClasses();
   }
 
+  /** Units in pixels to be set in x offset of the menu. */
+  @Input()
+  get xOffset() { return this._xOffset; }
+  set xOffset(value: number) {
+    let offset = +value;
+    if (isNaN(offset)) {
```

Comment:
Maybe use [`coerceNumberProperty `](https://github.com/angular/material2/blob/master/src/cdk/coercion/number-property.ts)? I guess it depends on how important throwing an error is.